tck/index.adoc

```diff
@@ -0,0 +1,51 @@
+= TCK Index
+
+The TCK is split into categories based on language constructs.
+The two main groups are clauses and expressions.
+Each group enumerates its members.
+Within each member, there is additional categories.
+
+There is also `uncategorized` directory containing uncategorized features.
+
+== Clauses
+
+Each clause has its own directory of Gherkin `.feature` files, or _feature_ for short.
+Each feature follows a naming convention of `<clause-name><group-id>.feature`.
+
+=== CREATE
+
+* `Create1`: Creating nodes
+** This group is about the creation of nodes.
+* `Create2`: Creating relationships
+** This group is about the creation of relationships.
+* `Create3`: ...
+** ...
+* `Create`_N_: Negative tests
+** This group is about failure cases for the `CREATE` clause; invalid queries.
```

Comment:
As I've said before (https://github.com/opencypher/openCypher/pull/342#issuecomment-461357008)

I do not think numbers is the appropriate way to sub-categorise, I think we should use names.

Reply:
I lean towards agreeing. While doing some of this categorization, I felt keeping the number stable is more complicated. Particular negative features, which where meant to occupy the last number in a category were troublesome, because you typically do not know in the beginning how many positive features you will have eventually in a category. Now that we plan to do the categorization in smaller steps on the main branch, this becomes more of an issue.

Reply:
Not disagreeing with either comment above. 

Just wanting to point out that what we will lose, with not using numbers, is the ability to specify ranges (of tests) that are applicable to some compliance level/feature group; i.e. we - and others - will be forced to rely on English names for black/whitelisting and full text enumeration of pertinent scenarios in other documents.

Reply:
The argument about ranges assumes that you can give a meaningful order to the feature. Doing so is not necessarily simple. If we accidentally pick the wrong order, we still need to enumerate numbers later. It is also overachieving, because you introduce more order than you may actually need eventually. Do compliance levels require to decide if scenarios on String or on integer literals come first?

Also, there are other ways to factor out definitions of compliance levels and such things. The TCK-API understands tags, as far as I can tell by now. So you can materialize compliance levels, etc. with tags and in documentation simply refer to the tag. That also allows a test runner to filter scenarios for such compliance levels very easily.